chromadb/utils/embedding_functions.py

```diff
@@ -385,3 +385,70 @@ def __call__(self, texts: Documents) -> Embeddings:
         if "predictions" in response:
             return response["predictions"]
         return []
+    
+
+class ResnetEmbeddingFunction(EmbeddingFunction):
+    def __init__(self, model_name: str = "resnet50"):
+        try:
+            self.torch = importlib.import_module("torch")
+        except ImportError:
+            raise ValueError(
+                "The Pytorch Python package is not installed. Please install it with 'pip install torch'."
+                )
+        try:
+            self.transforms = importlib.import_module("torchvision").transforms
+            self.models = importlib.import_module("torchvision").models
+        except ImportError:
+            raise ValueError(
+                "The torchvision Python packge is not installed. Please install it with 'pip install torchvision'"
+                )
+        try:
+            # Equivalent to from PIL import Image
+            self.Image = importlib.import_module("PIL").Image
+        except ImportError:
+            raise ValueError(
+                "The Pillow python package is not installed. Please install it with `pip install pillow`"
+            )
+        self.model_names = ["resnet18", "resnet34", "resnet50", "resnet101", "resnet153"]
+        if model_name not in self.model_names:
+            raise ValueError(f"Invalid model name. Allowed model names are: {', '.join(self.model_names)}")
+        self.vectors = []
+        self.preprocess = self.transforms.Compose(
+            [
+            self.transforms.Resize((224, 224)),
+            self.transforms.ToTensor(),
+            self.transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+            ]
+            )       
+        self.model = self.get_model(model_name) 
+        self.model.fc = self.torch.nn.Identity() 
+        self.model.eval()
+
+    def __call__(self, images: List[str | np.ndarray]) -> List[np.ndarray]:
```

Comment:
Doesn't conform to the `EmbeddingFunction` protocol.

https://github.com/sunilkumardash9/chroma/blob/37afa3f5eb78b673899f200d8295d7048815c680/chromadb/api/types.py#L85

1. We can only handle texts for now.
2. We should use Chroma's own types rather than numpy types.